src/main/java/backend/Logic.java

```diff
@@ -196,18 +191,36 @@ public Model getRepo(String repoId) {
      * @return True if label replacement on GitHub was a success, false otherwise.
      */
     public CompletableFuture<Boolean> replaceIssueLabels(TurboIssue issue, List<String> newLabels) {
+        eagerlyUpdateIssueLabels(issue, newLabels);
+
         logger.info("Changing labels for " + issue + " on GitHub");
-        return repoIO.replaceIssueLabels(issue, newLabels)
-                .thenApply(labels -> {
-                    logger.info("Changing labels for " + issue + " on UI");
-                    issue.setLabels(labels);
-                    refreshUI();
-                    return true;
-                })
+        return repoOpControl.replaceIssueLabels(issue, newLabels)
+                .thenApply(labels -> labels.containsAll(newLabels))
                 .exceptionally(Futures.withResult(false));
     }
 
     /**
+     * Assigns a list of labels to an issue offline and immediately reflect the change on the UI without
+     * concerning whether the labels have been updated on the remote repo's data source
+     *
+     * The UI will not be force refreshed if the set of labels are the same as the existing labels
+     * assigned to the issue.
```

Comment:
After another reading, I realized the code matches the description more than I realized at first. After some refactoring, the code is like this.

```java
/**
  * If there are changes to the labels, the issue and the UI are updated.
  */
  private void updateIssueLabelsAndUpdateUi(TurboIssue issue, List<String> labels) {
        
        boolean isLabelsUnchanged = issue.getLabels().size() == labels.size() && issue.getLabels().containsAll(labels);
        if (isLabelsUnchanged) { 
             logger.info("No changes to labels of " + issue + " on UI");
             return; 
        }

        logger.info("Changing labels for " + issue + " on UI");
        issue.setLabels(labels);
        refreshUI();

    }
```
It doesn't seem to be related to `eager update` concept. It just updates the issue and the UI. I prefer it did just one of those things. Combining the two things should be done at a higher level.